### PR TITLE
[AWS] Fix: Handle NoneType response in is_access_denied_exception

### PR DESCRIPTION
**Pull request by phalbert**
Fixes [PORT-10362](https://getport.atlassian.net/browse/PORT-10362)

### Summary

Addresses an issue where the `is_access_denied_exception` function in the AWS integration could raise an `AttributeError` if `e.response` is `None`. The function now includes a check to ensure `e.response` is not `None` before attempting to access its keys.

### Changes

- Added a check in `is_access_denied_exception` to ensure `e.response` is not `None` before accessing its keys.

## Type of change

Please leave one option from the following and delete the rest:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [x] Resync finishes successfully
- [x] Resync able to create entities
- [x] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [x] Integration able to create all default resources from scratch
- [x] Resync able to create entities
- [x] Resync able to update entities
- [x] Resync able to detect and delete entities
- [x] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

[PORT-10362]: https://getport.atlassian.net/browse/PORT-10362?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: integrations/aws/tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class MockException(Exception):
+    def __init__(self, response: Optional[Dict[str, Any]]) -> None:
+        self.response = response
+
+
+def test_access_denied_exception_with_response() -> None:
+    e = MockException(response={"Error": {"Code": "AccessDenied"}})
+    assert is_access_denied_exception(e) is True
```
Comment (lordsarcastic) [inline integrations/aws/tests/test_utils.py:12]:
```suggestion
    assert is_access_denied_exception(e)
```

File: integrations/aws/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def test_access_denied_exception_without_response() -> None:
+    e = MockException(response=None)
+    assert is_access_denied_exception(e) is False
```
Comment (lordsarcastic) [inline integrations/aws/tests/test_utils.py:17]:
```suggestion
    assert not is_access_denied_exception(e)
```

File: integrations/aws/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def test_access_denied_exception_with_other_error() -> None:
+    e = MockException(response={"Error": {"Code": "SomeOtherError"}})
+    assert is_access_denied_exception(e) is False
```
Comment (lordsarcastic) [inline integrations/aws/tests/test_utils.py:22]:
```suggestion
    assert not is_access_denied_exception(e)
```

File: integrations/aws/tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def test_access_denied_exception_no_response_attribute() -> None:
+    e = Exception("Test exception")
+    assert is_access_denied_exception(e) is False
```
Comment (lordsarcastic) [inline integrations/aws/tests/test_utils.py:27]:
```suggestion
    assert not is_access_denied_exception(e)
```

**Review by lordsarcastic — COMMENTED** [review]
Some succint fixes in the code

**Review by lordsarcastic — APPROVED** [review]

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**